hips/tiles/tests/test_tile.py

```diff
@@ -112,12 +124,26 @@ def test_read(self, pars):
         tile = self._read_tile(pars)
 
         assert tile.meta.order == pars['meta']['order']
+        assert tile.children[0].meta.order == pars['child1_meta']['order']
         assert isinstance(tile.raw_data, bytes)
 
         data = tile.data
+        child1_data = tile.children[0].data
         assert data.shape == pars['shape']
         assert data.dtype.name == pars['dtype']
-        assert_equal(tile.data[pars['pix_idx']], pars['pix_val'])
+        assert_equal(data[pars['pix_idx']], pars['pix_val'])
+
+        assert child1_data.shape == pars['child1_shape']
+        assert_equal(child1_data[pars['child1_pix_idx']], pars['child1_pix_val'])
+
+    @requires_hips_extra()
+    @pytest.mark.parametrize('pars', HIPS_TILE_TEST_CASES)
+    def test_children(self, pars):
+        tile = self._read_tile(pars)
+        child1_data = tile.children[0].data
```

Comment:
Here I would suggest to put this on a separate line:
```
child_tiles = tile.children
```
Then you can on the following lines access meta and data from each child as you like, e.g. you could do:
```
ipix = [_.meta.ipix for _ in child_tiles]
```
and then assert on this `ipix` list, to establish the current behaviour of the function about the order of child tiles returned.